Tasks/JavaToolInstallerV0/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -27,31 +28,32 @@ export class JavaFilesExtractor {
         }
     }
 
-    private async unzipExtract(file, destinationFolder) {
-        console.log(taskLib.loc('UnzipExtractFile', file));
-        if (process.platform === 'win32') {
-            // build the powershell command
-            const escapedFile = file.replace(/'/g, "''").replace(/"|\n|\r/g, ''); // double-up single quotes, remove double quotes and newlines
-            const escapedDest = destinationFolder.replace(/'/g, "''").replace(/"|\n|\r/g, '');
-            const command: string = `$ErrorActionPreference = 'Stop' ; try { Add-Type -AssemblyName System.IO.Compression.FileSystem } catch { } ; [System.IO.Compression.ZipFile]::ExtractToDirectory('${escapedFile}', '${escapedDest}')`;
-    
-            // change the console output code page to UTF-8.
-            const chcpPath = path.join(process.env.windir, "system32", "chcp.com");
-            await taskLib.exec(chcpPath, '65001');
-    
-            // run powershell
-            const powershell: tr.ToolRunner = taskLib.tool('powershell')
-                .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
-                .arg(command);
-            await powershell.exec();
-        } else {
-            const unzip: tr.ToolRunner = taskLib.tool('unzip')
-                .arg(file);
-            await unzip.exec(<tr.IExecOptions>{ cwd: destinationFolder });
-        }
+    private isTar(file) {
+        var name = this.win ? file.toLowerCase() : file;
+        // standard gnu-tar extension formats with recognized auto compression formats
+        // https://www.gnu.org/software/tar/manual/html_section/tar_69.html
+        return name.endsWith('.tar')      // no compression
+            || name.endsWith('.tar.gz')   // gzip
+            || name.endsWith('.tgz')      // gzip
+            || name.endsWith('.taz')      // gzip
+            || name.endsWith('.tar.Z')    // compress
+            || (this.win && name.endsWith('tar.z')) // no case comparison for win
```

Comment:
Even on *nix, do we care about case sensitivity?  Going case-insensitive all the time means we don't have to special-case things for Windows.  Since even on *nix I don't think there's a risk of a collision on one of these file types, I think it reduces the chance of having a platform-dependent bug